actions/ConnectivityAction.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * This file is part of RSS-Bridge, a PHP project capable of generating RSS and
+ * Atom feeds for websites that don't have one.
+ *
+ * For the full license information, please view the UNLICENSE file distributed
+ * with this source code.
+ *
+ * @package	Core
+ * @license	http://unlicense.org/ UNLICENSE
+ * @link	https://github.com/rss-bridge/rss-bridge
+ */
+
+/**
+ * Checks if the website for a given bridge is reachable.
+ *
+ * **Remarks**
+ * - This action is only available in debug mode.
+ * - Returns the bridge status as Json-formatted string.
+ * - Returns an error if the bridge is not whitelisted.
+ * - Returns a responsive web page that automatically checks all whitelisted
+ * bridges (using JavaScript) if no bridge is specified.
+ */
+class ConnectivityAction extends ActionAbstract {
+	public function execute() {
+
+		if(!Debug::isEnabled()) {
+			returnError('This action is only available in debug mode!');
+		}
+
+		if(!isset($this->userData['bridge'])) {
+			$this->returnEntryPage();
+			return;
+		}
+
+		$bridgeName = $this->userData['bridge'];
+
+		if(!Bridge::isWhitelisted($bridgeName)) {
+			header('Content-Type: text/html');
+			returnServerError('Bridge is not whitelisted!');
+		}
+
+		$this->reportBridgeConnectivity($bridgeName);
+
+	}
+
+	/**
+	 * Generates a report about the bridge connectivity status and sends it back
+	 * to the user.
+	 *
+	 * The report is generated as Json-formatted string in the format
+	 * {
+	 *   "bridge": "<bridge-name>",
+	 *   "successful": true/false
+	 * }
+	 *
+	 * @param string $bridgeName Name of the bridge to generate the report for
+	 * @return void
+	 */
+	private function reportBridgeConnectivity($bridgeName) {
+
+		header('Content-Type: text/json');
+
+		$retVal = array(
+			'bridge' => $bridgeName,
+			'successful' => false,
+			'http_code' => 200,
+		);
+
+		$bridge = Bridge::create($bridgeName);
+
+		if($bridge === false) {
+			echo json_encode($retVal);
+			return;
+		}
+
+		$curl_opts = array(
+			CURLOPT_CONNECTTIMEOUT => 5
+		);
+
+		try {
+			$reply = getContents($bridge::URI, array(), $curl_opts, true);
+
+			if($reply) {
+				$retVal['successful'] = true;
+				if (isset($reply['header'])) {
+					if (strpos($reply['header'], 'HTTP/1.1 301 Moved Permanently') !== false) {
+						$retVal['http_code'] = 301;
+					}
+				}
+			}
+		} catch(Exception $e) {
+			$retVal['successful'] = false;
+		}
+
+		echo json_encode($retVal);
+
+	}
+
+	private function returnEntryPage() {
+	echo <<<EOD
+<!DOCTYPE html>
+
+<html>
+	<head>
+		<link
+			rel="stylesheet"
+			href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
```

Comment:
I am not sure about linking to bootstap's CDNs tbh. If needed I'll do a CSS in another PR

Reply:
Do you mean to include the CSS files in the repository, or to replace Bootstrap entirely?

Reply:
Maybe include the CSS for now